src/main/java/com/google/sps/servlets/AddSaleCardServlet.java

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.sps.data.HttpServletUtils;
+import java.io.IOException;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet.
+ */
+@WebServlet("/add-saleCard")
+public class AddSaleCardServlet extends HttpServlet {
+
+  private final static float MAX_DISTANCE = 3000f; // 3 kilometers
+  private final static float MIN_DISTANCE = 0f;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: Get if vendor is logged in with auth API
+    boolean vendorStatus = false;
+    
+    if (vendorStatus) {
+      // -- The way we get these values might change. This is just an idea ---------------
+      // Get parameters
+      String businessName = HttpServletUtils.getParameter(request, "businessName", "");
```

Comment:
Just curiosity about HttpServletUtils, does not `request.getParameter("businessName");` work?

Reply:
Yes, `request.getParameter` does work. It is actually called inside `HttpServletUtils.getParameter`. I use `HttpServletUtils` to provide a default value (the third parameter) in case there isn't a request's parameter or it is empty.